app/models/search_stat.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class SearchStat < ApplicationRecord
+  validates :keyword, presence: true
+  validates :raw_response, presence: true
```

Comment:
We can validate many fields on the same line.

```suggestion
  belongs_to :user
  
  validates :keyword, :ad_count, :link_count, :total_result_count, :raw_response, presence: true
```

```ruby
    create_table :search_stats do |t|
      t.string :keyword, null: false
      t.integer :ad_count, null: false, default: 0
      t.integer :link_count, null: false, default: 0
      t.bigint :total_result_count, null: false, default: 0
      t.text :raw_response, null: false
      t.references :user, null: false, foreign_key: true

      t.timestamps
    end
```

Reply:
Initially rows will be created with the column `keyword` only. After scrapping is done it will find rows by id and fill in other columns. So, I should really only validate `:keyword` right?